common/src/main/java/io/netty/util/internal/PlatformDependent0.java

```diff
@@ -198,6 +214,20 @@ static void throwException(Throwable cause) {
         UNSAFE.throwException(checkNotNull(cause, "cause"));
     }
 
+    static boolean hasDirectBufferNoCleanerConstructor() {
+        return DIRECT_BUFFER_CONSTRUCTOR != null;
+    }
+
+    static ByteBuffer allocateDirectNoCleaner(int capacity) {
+        assert DIRECT_BUFFER_CONSTRUCTOR != null;
+        long address = UNSAFE.allocateMemory(capacity);
+        try {
+            return (ByteBuffer) DIRECT_BUFFER_CONSTRUCTOR.newInstance(address, capacity);
+        } catch (Throwable cause) {
+            throw new IllegalStateException(cause);
```

Comment:
Throw an `Error` if you don't expect this to fail. Otherwise it should be handled properly. e.g:

- Unwrap `InvocationTargetException` and rethrow the cause.
- Throw an `Error` for other impossible exceptions such as `IllegalAccessException` and `InstantiationException`.